### ISSUE TITLE
Problems with sync across devices

### ISSUE THREAD

**Issue by duckmanBR** [DestinyItemManager/DIM#2223]
I'll try to be concise, but it'll be hard.

beta on mobile, everything ok, local and drive shows same amount of information
![screenshot_20170924-101735](https://user-images.githubusercontent.com/32077894/30783006-d32b7e2c-a112-11e7-8a27-efeb5bea9b63.jpg) <- on mobile

access beta on computer for the first time (or all DIM cookies removed):
if you don't have Drive automatically, login/authorize and then everything will work (but will show that nothing is saved locally (you need to change something to show that its saved local)
![clipboard01](https://user-images.githubusercontent.com/32077894/30783037-542155ce-a113-11e7-9a15-60bb041e82b5.jpg) <- on desktop
everything works fine (you only need to reload the page on the other device after a modification, didn't try to wait to see if there's any auto-sync)

**BUT**
if you had Drive automatically logged (don't know how it happens, but it was my default behavior), it won't find anything:
![clipboard02](https://user-images.githubusercontent.com/32077894/30783111-29096fc4-a114-11e7-8fee-05af3003c673.jpg) <- on desktop

and if you reload on your other device, it'll look like a data wipe, but if you go to the storage page, the local data is there:
![screenshot_20170924-103746](https://user-images.githubusercontent.com/32077894/30783167-c0b758cc-a114-11e7-8da7-f50461341b8c.jpg) <- on mobile
but then you only be able to see it if you logout Drive.  but if you login back it will show the (empty) data from Drive (even after refresh).  If you force sync, it will download the nothing from drive, if you edit something locally, it'll upload only the modification.  Couldn't find a way to send those local data to drive.

unfortunately I don't have the time now to do this test on reverse (first on desktop and then on mobile), but I hope that this info helps solve what's happening.

PS: before I did this tests I forgot to make a new backup, so I lost all my settings, but I'm in beta so it was bound to happen :D

**Comment by Korben85** [comment]
There is seriously something wrong with the sync.
When I switch between D1 and D2 for example, all my notes are gone. 
When I load a Dim.json file, the data is loaded but when I add a note in D2 and then go to the data page, only the notes for D2 are left and all notes from D1 are gone.
Switching from D1 to D2 or vise versa ends up in having no notes at all....

**Comment by bhollis** [comment]
@Korben85 you're describing what is I think a separate bug, which is present in release but fixed in beta. The fix will go out to release tonight. @duckmanBR is describing something else.

**Comment by Korben85** [comment]
Of course you are right. Just wanted to correct my post cause I checked the beta and the dev version and there it's fixed.

**Comment by Sleepydragn1** [comment]
I can replicate this as well, even on the Beta version. To clarify the OP, which is a little bit messy:

When Google Drive is connected to a DIM account, signing in on a completely new browser (i.e. just signing out from an existing session doesn't cause any issues) seems to sync the new (blank) local session data from that new browser to Google Drive, overriding any existing data on Google Drive. 

On the user end of things, DIM on the original browser will use Google Drive's data instead of the local data (even if exported), causing it to seem like everything has been lost, even though their local storage data remains intact. If they attempt to log out of Google Drive, and then log back in, or if they click force sync data (both of which are common troubleshooting steps), the blank data overrides their local storage data. This is intended behavior, but might still cause issues/frustration.

The workaround for users is as follows:
1. On your PC (or whatever your original device was) log out of Google Drive from the DIM Storage page. 
2. Go to Google Drive, click the Settings cogwheel in the top right, and go to Manage Apps. Find Destiny Item Manager, and in the Options dropdown, click Delete hidden app data. **If you don't do this, DIM will delete your local data when it syncs with the empty storage file on Google Drive.**
3. On your PC, log back into Google Drive. It should now say "No data" rather than "0 loadouts, 0 items with tags or notes, etc."
4. Click Force sync data at the top.

Regardless, from my uninformed perspective, it seems like the fix is to sync from Google Drive to local storage first whenever setting up a new browser login. 

In the future renaming "Force sync data" to "Force sync from Google Drive to local storage" or something similar and/or adding in a "Force sync from local storage to Google Drive" might mitigate any future issues.

**Comment by bhollis** [comment]
@Sleepydragn1 I don't doubt that this is happening to you, and I'm sorry that you're experiencing it. Unfortunately, the intended behavior as written is basically what you suggest, so there must be some more subtle bug going on.

**Comment by bhollis** [comment]
@Sleepydragn1 @duckmanBR is this still happening? 4.18 includes a change that should help.

**Comment by duckmanBR** [comment]
Ok, I'll try later today.

**Comment by Sleepydragn1** [comment]
@bhollis 4.18 seems to fix the issue. I had to log into Google Drive on every new browser set up, but on Chromium and Firefox it worked flawlessly after that.

Oddly enough, though, the first time I pulled up DIM on my mobile device (which I had been previously using to test this as a second device) it deleted everything in the Google Drive storage. After I did the workaround I posted before, pulling up DIM on mobile again didn't delete anything like before, but it wasn't properly syncing regardless. Not sure if that's an issue or not.
